### PR TITLE
webui: Fix a bug where collapsing the sidebar clears the search results.

### PR DESCRIPTION
**Pull request by junhaoliao**
# References
Internally it was found that collapsing / expanding the sidebar would clear the search results. Further investigation found that the search page was re-rendered when the `isSidebarCollapsed` state variable got updated, causing unnecessary and undesired re-render of page contents. 

It was also identified that the pseudo user authentication system is no longer in use and can be removed.

# Description
1. Remove user login methods and related UI components.
2. Rename React root component to `root` which is a more standard convention.
 
# Validation performed
1. `cd <PROJECT_ROOT>; task`
2. `cd ./build/clp-package/sbin; ./start-clp.sh`
3. `./compress.sh ~/samples/hive-24hr/i-00c90a0f/`
4. Opened the WebUI address in a browser.
5. Started a query with string 123 and observed the job finished and the results are displayed in the results table.
6. Clicked the "Collapse" button to collapse the sidebar and observed the results still present within the search page.
7. Clicked the "Expand" button to expand the sidebar and observed the results still present within the search page.

### CODE REVIEW

**Comment by junhaoliao** [comment]
Please put this review on hold. Let me also remove the unused user authentication system in this PR instead.

**Comment by junhaoliao** [comment]
The PR descriptions have been updated. Please resume the review.

File: components/webui/imports/ui/App.jsx [hunk 1]
```diff
@@ -29,7 +28,6 @@ const ROUTES = [
 ];
 
 const App = () => {
-    const [loggedIn, setLoggedIn] = React.useState(false);
```
Comment (kirkrodrigues) [inline components/webui/imports/ui/App.jsx:32]: Also remove calls to `login` and `setLoggedIn`

**Review by kirkrodrigues — CHANGES_REQUESTED** [review]

**Review by kirkrodrigues — APPROVED** [review]
How about:

webui: Don't use an expression to render routes to avoid unnecessary re-renders. (#336)

(At least that's my understanding of the issue.)

**Comment by junhaoliao** [comment]
> How about:
> 
> webui: Don't use an expression to render routes to avoid unnecessary re-renders. (#336)
> 
> (At least that's my understanding of the issue.)

JSX itself is an expression, but sometimes React is smart enough to avoid re-evaluation of the JSX. The [initial fix](https://github.com/y-scope/clp/pull/336/commits/864f5111816d645ba5b6ae673ea7cdf623904833) does exactly that although the `routeSwtich` constant was later inlined. Supposedly another fix is to keep the `<Routes/>` as a component and move it out of the `<App/>`.

It might be more clear to say `webui: Avoid unnecessary React route-switch re-render within functional components. (#336)`. What do you think?

**Pull request merged**